pkg/build/util/util.go

```diff
@@ -75,7 +76,30 @@ func IsBuildComplete(build *buildapi.Build) bool {
 }
 
 // BuildNameForConfigVersion returns the name of the version-th build
-// for the config that has the provided name
+// for the config that has the provided name.
 func BuildNameForConfigVersion(name string, version int) string {
 	return fmt.Sprintf("%s-%d", name, version)
 }
+
+// ConfigNameForBuild returns the name of the build config from a
+// build name.
+func ConfigNameForBuild(name string) string {
+	parts := strings.Split(name, "-")
+	if len(parts) != 2 {
+		return ""
+	}
+	return parts[0]
+}
+
+// VersionForBuild returns the version from the provided build name
+func VersionForBuild(name string) int {
+	parts := strings.Split(name, "-")
```

Comment:
Actually you store the version in the annotations, more preferrable than parsing. Fixed.